File: code/backend/php/tests/Unit/Core/Service/Matches/PropertySystemModeServiceTest.php
<?php

declare(strict_types=1);

namespace Games\Paths\Tests\Unit\Core\Service\Matches;

use Games\Paths\Core\Service\Matches\PropertySystemModeService;
use PHPUnit\Framework\TestCase;

class PropertySystemModeServiceTest extends TestCase
{
    public function testMaintenanceTrueWhenStatusMatches(): void
    {
        $this->assertTrue((new PropertySystemModeService('MAINTENANCE'))->isMaintenance());
        $this->assertTrue((new PropertySystemModeService('maintenance'))->isMaintenance());
        $this->assertTrue((new PropertySystemModeService(' Maintenance '))->isMaintenance());
    }

    public function testMaintenanceFalseForOtherStatus(): void
    {
        $this->assertFalse((new PropertySystemModeService('OK'))->isMaintenance());
        $this->assertFalse((new PropertySystemModeService(''))->isMaintenance());
    }
}

File: code/backend/java/core/src/test/java/games/paths/core/service/match/PropertySystemModeServiceTest.java
package games.paths.core.service.match;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class PropertySystemModeServiceTest {

    @Test
    void maintenanceTrueWhenStatusEqualsMaintenanceCaseInsensitive() {
        assertTrue(new PropertySystemModeService("MAINTENANCE").isMaintenance());
        assertTrue(new PropertySystemModeService("maintenance").isMaintenance());
        assertTrue(new PropertySystemModeService("MainTenance").isMaintenance());
    }

    @Test
    void maintenanceFalseForOtherStatus() {
        assertFalse(new PropertySystemModeService("OK").isMaintenance());
        assertFalse(new PropertySystemModeService("").isMaintenance());
        assertFalse(new PropertySystemModeService(null).isMaintenance());
    }
}
